packages/eslint-plugin/src/utils/create-no-exported-rule/is-styled-component.ts

```diff
@@ -0,0 +1,59 @@
+import type { Rule } from 'eslint';
+import type { MemberExpression } from 'estree';
+
+import { findCompiledImportDeclarations } from '../ast';
+
+type Node = Rule.Node;
+type RuleContext = Rule.RuleContext;
+
+/**
+ * Returns the MemberExpression node that could define a Compiled component
+ *
+ * @param nodes
+ * @returns
+ */
+const findMemberExpressionNode = (nodes: Node[]): MemberExpression | undefined => {
+  const node = nodes.find(
+    (n) => n.type === 'TaggedTemplateExpression' || n.type === 'CallExpression'
+  );
+
+  // Eg. const Component = styled.button(style)
+  if (node && node.type === 'CallExpression' && node.callee.type === 'MemberExpression') {
+    return node.callee;
+  }
+
+  // Eg. const Component = styled.div`${styles}`;
+  if (node && node.type === 'TaggedTemplateExpression' && node.tag.type === 'MemberExpression') {
+    return node.tag;
+  }
+
+  return;
```

Comment:
What about `styled(button)(styled)`?

Reply:
Can we do this in Compiled?

Reply:
oh,sry, what i mean is

```
const button = styled.div({ color: 'red' });
const button2 = styled(button)({ width: '100px' })
```

Reply:
I think I get it now. I'll check both cases

1. exporting a component that extends a component using css. Eg

```
const button = styled.div(css({ color: 'red' }));
export const button2 = styled(button)({ width: '100px' });
```

OR 

2. exporting a component that extends another component and has a css reference
```
const button = styled.div({ color: 'red' });
export const button2 = styled(button)(css({ width: '100px' }));
```

Reply:
That's a nice summary 👍

Reply:
they should be the same in terms of AST